### ISSUE TITLE
Remove unnecessary conditions from code

### ISSUE THREAD

**Issue by jonkoops** [keycloak/keycloak#17770]
In various places there is code that has [unnecessary conditions](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md). This leads to a poor implementation and incorrect types and should not occur. 

The following needs to happen before this issue can be closed:
1. Refactor code that is flagged by the `@typescript-eslint/no-unnecessary-condition` rule so it no longer produces warnings (this can be split up in multiple PRs).
2. Change the same rule to  the `error` level so linting fails when unnecessary conditions are introduced.

**Comment by Carlvensky** [comment]
Any update on this issue? Was it fixed?

**Comment by jonkoops** [comment]
Nope, nothing has changed since this issue was reported. If someone wants to volunteer to take this we can assign it to that person.

**Comment by datasciencetrialgit** [comment]
Can I take this issue and work on it?

**Comment by shawkins** [comment]
> Can I take this issue and work on it?

@datasciencetrialgit please open a PR. If it looks like you are able to make headway we can formally assign this to you.
